lib/kernel/src/socket.erl

```diff
@@ -2769,9 +2775,18 @@ Equivalent to [`open(Domain, Type, Protocol, #{})`](`open/4`).
 
 Equivalent to [`open(Domain, Type, default, #{})`](`open/4`).
 """.
--spec open(Domain :: term(), Type :: term(), Opts :: map()) -> _;
-          (Domain :: term(), Type :: term(), Protocol :: term()) -> _.
-
+-spec open(Domain, Type, Opts) -> {'ok', Socket} | {'error', Reason} when
```

Comment:
It doesn't seem that there is any need in overloaded specs, the same can be expresses via a single spec (until I am missing something).

```erlang
-spec open(Domain, Type, Opts | Protocol) -> {'ok', Socket} | {'error', Reason} when
      Domain :: domain() | integer(),
      Type :: type() | integer(),
      Opts :: map(),
      Protocol :: default | protocol() | integer()
      Socket :: socket(),
      Reason :: posix() | protocol.
```

Reply:
Good catch!